Description: minimize twisted dependencies
Author: Julian Taylor <jtaylor.debian@googlemail.com>
Forwarded: not-needed
Index: foolscap.git/setup.py
===================================================================
--- foolscap.git.orig/setup.py	2011-06-15 18:11:07.168005624 +0200
+++ foolscap.git/setup.py	2011-06-15 18:11:11.838005620 +0200
@@ -71,7 +71,9 @@
             "flappserver = foolscap.appserver.cli:run_flappserver",
             "flappclient = foolscap.appserver.client:run_flappclient",
             ] }
-    setup_args['install_requires'] = ['twisted >= 2.4.0']
+    setup_args['install_requires'] = ['twisted-core >= 2.4.0',
+                                      'twisted-web >= 2.4.0',
+                                      'zope.interface']
     setup_args['extras_require'] = { 'secure_connections' : ["pyOpenSSL"] }
     # note that pyOpenSSL-0.7 and recent Twisted causes unit test failures,
     # see bug #62
